service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -150,139 +176,136 @@ public void reset(final PolicyEvaluationRequestCacheKey key) {
         delete(keys);
     }
 
-    @Override
-    public void resetForPolicySet(final String zoneId, final String policySetId) {
-        String key = policySetKey(zoneId, policySetId);
+    private void resetForEntity(final String zoneId, final String entityId, final EntityType entityType,
+            final BiFunction<String, String, String> getKey) {
+        String key = getKey.apply(zoneId, entityId);
         String timestamp = timestampValue();
+        logSetEntityTimestampsDebugMessage(timestamp, key, entityId, entityType);
+        set(key, timestamp);
+    }
+
+    private void setEntityIfNotExists(final String zoneId, final String entityId,
+            final BiFunction<String, String, String> getKey) {
+        String key = getKey.apply(zoneId, entityId);
+        setIfNotExists(key, timestampValue());
+    }
+
```

Comment:
setEntityIfNotExists will set the timestamp for the object if the object is not in the cache. resetForEntity will reset the timestamp regardless if the value if the value is in the cache